pkg/controller/gcp/gcp.go

```diff
@@ -17,38 +17,54 @@ limitations under the License.
 package gcp
 
 import (
-	"sigs.k8s.io/controller-runtime/pkg/manager"
+	ctrl "sigs.k8s.io/controller-runtime"
 
 	"github.com/crossplaneio/crossplane/pkg/controller/gcp/cache"
 	"github.com/crossplaneio/crossplane/pkg/controller/gcp/compute"
 	"github.com/crossplaneio/crossplane/pkg/controller/gcp/database"
 	"github.com/crossplaneio/crossplane/pkg/controller/gcp/storage"
 )
 
-func init() {
-	// AddToManagerFuncs is a list of functions to create controllers and add them to a manager.
-	AddToManagerFuncs = append(AddToManagerFuncs,
-		cache.Add,
-		cache.AddClaim,
-		compute.Add,
-		compute.AddClaim,
-		database.Add,
-		database.AddPostgreSQLClaim,
-		database.AddMySQLClaim,
-		storage.Add,
-		storage.AddClaim,
-	)
-}
+// Controllers passes down config and adds individual controllers to the manager.
+type Controllers struct{}
 
-// AddToManagerFuncs is a list of functions to add all Controllers to the Manager
-var AddToManagerFuncs []func(manager.Manager) error
+// SetupWithManager adds all GCP controllers to the manager.
+func (c *Controllers) SetupWithManager(mgr ctrl.Manager) error {
+	if err := (&cache.CloudMemorystoreInstanceClaimController{}).SetupWithManager(mgr); err != nil {
+		return err
```

Comment:
Did we have any particular reason to fail fast for these errors? As a developer I usually find it to be a better experience if I can get back a list of all the errors at once, instead of having the first error hide other ones. In this case, that'd mean collecting all the `err`s and returning them all together as a group. Or aggregated under a single error.

Reply:
That is a valid point. I think the logic is that there is no reason to try adding additional controllers when we are certain we will fail. However, this is in-line with how we currently add controllers with the `AddToManager` pattern (return `err` as soon as one is encountered). A change would likely be out of the scope of this PR